### PR TITLE
Increase upper limit requirement on phpstan

### PR DESCRIPTION
**Pull request by juliangut**
PHPStan 0.12 have just been released

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -17,12 +17,12 @@
     ],
     "require": {
         "php": "^7.1",
-        "phpstan/phpstan": "^0.11"
+        "phpstan/phpstan": "~0.11"
```
Comment (mremi) [inline composer.json]: Better to keep `^` operator for versions 0.* because of potential BC breaks, which is the case here (see Travis output).

Comment (juliangut) [reply composer.json]: ^0.11 prevents installation of phpstan 0.12

Comment (mremi) [reply composer.json]: Yep, but you can use `^0.12`

Comment (juliangut) [reply composer.json]: well, this change should not break BC with 0.11.* and so I intended to keep it, maybe "^0.11|^0.12" what do you think?

File: .travis.yml [hunk 2]
```diff
@@ -20,8 +21,6 @@ cache:
   directories:
     - $HOME/.composer/cache/files
 
-before_install: composer self-update
```
Comment (mremi) [inline .travis.yml]: Why?

File: composer.json [hunk 3]
```diff
@@ -17,12 +17,12 @@
...
     "require": {
         "php": "^7.1",
-        "phpstan/phpstan": "^0.11"
+        "phpstan/phpstan": "^0.12"
     },
     "require-dev": {
         "friendsofphp/php-cs-fixer": "^2.12",
         "localheinz/composer-normalize": "^1.3",
-        "phpstan/phpstan-phpunit": "^0.11",
+        "phpstan/phpstan-phpunit": "~0.11",
```
Comment (mremi) [inline composer.json]:
```suggestion
        "phpstan/phpstan-phpunit": "^0.12",
```

File: .travis.yml (hunk 2 above)
Comment (juliangut) [reply .travis.yml]: Duplicated, it is already in the line below

**Comment by juliangut** [comment]
When you merge please squash, there is no point on having this much commits with minor changes

**Comment by juliangut** [comment]
Any luck on merging this?

**Comment by mremi** [comment]
Could you please update CHANGELOG.md?

**Comment by juliangut** [comment]
Updated, not sure if is what you were expecting

**Comment by mremi** [comment]
Thanks, [new tag](https://github.com/ekino/phpstan-banned-code/releases/tag/v0.3.0) is available

**Pull request merged**